source/extensions/quic_listeners/quiche/envoy_quic_dispatcher.cc

```diff
@@ -51,11 +51,17 @@ std::unique_ptr<quic::QuicSession> EnvoyQuicDispatcher::CreateQuicSession(
     quic::QuicConnectionId server_connection_id, const quic::QuicSocketAddress& self_address,
     const quic::QuicSocketAddress& peer_address, absl::string_view /*alpn*/,
     const quic::ParsedQuicVersion& version) {
+  quic::QuicConfig quic_config = config();
+  // TODO(danzh) setup flow control window via config.
+  quic_config.SetInitialStreamFlowControlWindowToSend(
+      Http2::Utility::OptionsLimits::MIN_INITIAL_STREAM_WINDOW_SIZE);
```

Comment:
Ditto on default rather than initial.
Could this have contributed to the tsan failures?  Maybe see if changing this back means you can revert the wait time?